test/conftest.py

```diff
@@ -66,3 +99,83 @@ def pytest_collection_modifyitems(config, items):
             if run_id:
                 item._nodeid = f'{item._nodeid}.{run_id}'
                 item.name = f'{item.name}.{run_id}'
+
+
+def prepare_dir(dirname: str, pattern: str) -> None:
+    # Ensure the dir exists
+    pathlib.Path(dirname).mkdir(parents=True, exist_ok=True)
+    # Remove old artifacts
+    for p in glob.glob(os.path.join(dirname, pattern), recursive=True):
+        pathlib.Path(p).unlink()
+
+
+def prepare_dirs(tempdir_base: str, modes: list[str]) -> None:
+    prepare_dir(tempdir_base, "*.log")
+    for mode in modes:
+        prepare_dir(os.path.join(tempdir_base, mode), "*.log")
+        prepare_dir(os.path.join(tempdir_base, mode), "*.reject")
+        prepare_dir(os.path.join(tempdir_base, mode, "xml"), "*.xml")
+        shutil.rmtree(os.path.join(tempdir_base, mode, "failed_test"), ignore_errors=True)
+        prepare_dir(os.path.join(tempdir_base, mode, "failed_test"), "*")
+        prepare_dir(os.path.join(tempdir_base, mode, "allure"), "*.xml")
+        if TEST_RUNNER == "pytest":
+            shutil.rmtree(os.path.join(tempdir_base, mode, "pytest"), ignore_errors=True)
+            prepare_dir(os.path.join(tempdir_base, mode, "pytest"), "*")
+
+
+@universalasync.async_to_sync_wraps
+async def start_s3_mock_services(minio_tempdir_base: str) -> None:
```

Comment:
There is no reason why this large collection of utility functions, and all their imports, need to be in conftest.py. It can be in a separate file, pylib/s3_mock.py or whatever you might want to call it.

Reply:
These functions are candidates to be fixtures when we completely switch to pytest. Even if we'll decide to stay with the hooks, they will be used just in this file. So, they probably found their place :)

But in general you're right: it's better to keep any reusable code out of `conftest.py`